core/common/src/annotation/TextBlock.ts

```diff
@@ -816,6 +1096,113 @@ export class TextBlock extends TextBlockComponent {
 
     if (!marginsAreEqual) return false;
 
-    return this.paragraphs.every((paragraph, index) => paragraph.equals(other.paragraphs[index]));
+    if (this.children && other.children) {
+      if (this.children.length !== other.children.length) {
+        return false;
+      }
+
+      return this.children.every((child, index) => other.children && child.equals(other.children[index]));
+    }
+
+    return true;
+  }
+}
+
+/**
+ * Recursively traverses a tree of TextBlockComponent objects, yielding each component along with its parent container.
+ * This generator enables depth-first iteration over all components in a text block structure, including paragraphs, lists, and runs.
+ *
+ * @param block The root TextBlockComponent to start traversal from.
+ * @param parent (Optional) The parent ContainerComponent of the current block. For the root, this is undefined.
+ * @returns An IterableIterator yielding objects with the current component and its parent container.
+ * @beta
```

Comment:
You omitted a release tag, so it defaulted to `@public`. It can be at most `@beta` because `TextBlockComponent` is `@beta`. Is that what you want? (I added the tag to make extract-api succeed so I could see a summary of your API changes).

Reply:
Beta is good

Reply:
also yeah, I haven't run extract-api or lint yet